plugins/content_distribution/providers/cortex/lib/api/KalturaCortexApiDistributionJobProviderData.php

```diff
@@ -0,0 +1,183 @@
+			$flavorAsset = reset($flavorAssets);
+		else // take the source asset
+			$flavorAsset = assetPeer::retrieveOriginalReadyByEntryId($distributionJobData->entryDistribution->entryId);
+		
+		if($flavorAsset) 
+		{
+			$syncKey = $flavorAsset->getSyncKey(flavorAsset::FILE_SYNC_FLAVOR_ASSET_SUB_TYPE_ASSET);
+			if(kFileSyncUtils::fileSync_exists($syncKey))
+			{
+				$this->videoAssetFilePath = kFileSyncUtils::getLocalFilePathForKey($syncKey, false);
+			}
+		}
+		
+		$thumbAssets = assetPeer::retrieveByIds(explode(',', $distributionJobData->entryDistribution->thumbAssetIds));
+		if(count($thumbAssets))
+		{
+			$syncKey = reset($thumbAssets)->getSyncKey(thumbAsset::FILE_SYNC_FLAVOR_ASSET_SUB_TYPE_ASSET);
+			if(kFileSyncUtils::fileSync_exists($syncKey))
+				$this->thumbAssetFilePath = kFileSyncUtils::getLocalFilePathForKey($syncKey, false);
+		}
+		
+		$this->addCaptionsData($distributionJobData);
+	}
+	
+	private static $map_between_objects = array
+	(
+		"videoAssetFilePath",
+		"thumbAssetFilePath",
+		"captionsInfo",
+	);
+
+	public function getMapBetweenObjects ( )
+	{
+		return array_merge ( parent::getMapBetweenObjects() , self::$map_between_objects );
+	}
+	
+	private function addCaptionsData(KalturaDistributionJobData $distributionJobData)
+	{
+		/* @var $mediaFile KalturaDistributionRemoteMediaFile */
+		$assetIdsArray = explode ( ',', $distributionJobData->entryDistribution->assetIds );
+		if (empty($assetIdsArray)) return;
+		$assets = array ();
+		$this->captionsInfo = new KalturaCortexApiCaptionDistributionInfoArray();
+		
+		foreach ( $assetIdsArray as $assetId ) {
+			$asset = assetPeer::retrieveByIdNoFilter( $assetId );
+			if (!$asset)
+			{
+				KalturaLog::err("Asset [$assetId] not found");
+				continue;
+			}
+			if ($asset->getStatus() == asset::ASSET_STATUS_READY) {
+				$assets [] = $asset;
+			}
+			else{
+				KalturaLog::err("Asset [$assetId] has status [".$asset->getStatus()."]. not added to provider data");
+			}
+		}
+
+		foreach ( $assets as $asset ) {
+			$assetType = $asset->getType ();
+			switch ($assetType) {
+				case CaptionPlugin::getAssetTypeCoreValue ( CaptionAssetType::CAPTION ):
+					$syncKey = $asset->getSyncKey ( asset::FILE_SYNC_FLAVOR_ASSET_SUB_TYPE_ASSET );
+					if (kFileSyncUtils::fileSync_exists ( $syncKey )) {
+						$captionInfo = $this->getCaptionInfo($asset, $syncKey, $distributionJobData);
+						if ($captionInfo)
+						{
+							$captionInfo->label = $asset->getLabel();
+							if(!$captionInfo->label)
+								$captionInfo->label = $asset->getLanguage();
+							$captionInfo->language = $this->getLanguageCode($asset->getLanguage());
+							if ($captionInfo->language)
+								$this->captionsInfo [] = $captionInfo;
+							else
+								KalturaLog::err('The caption ['.$asset->getId().'] has unrecognized language ['.$asset->getLanguage().']'); 
+						}
+					}
+					break;
+				case AttachmentPlugin::getAssetTypeCoreValue ( AttachmentAssetType::ATTACHMENT ) :
```

Comment:
attachment plugin is also dependency for cortex?

Reply:
No, fixed here - https://github.com/kaltura/server/commit/d651a176b5e566632633bfc51e76cb14f57af4d1